plotter/field.go

```diff
@@ -0,0 +1,188 @@
+// Copyright ©2019 The Gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package plotter
+
+import (
+	"math"
+
+	"gonum.org/v1/plot"
+	"gonum.org/v1/plot/vg"
+	"gonum.org/v1/plot/vg/draw"
+)
+
+// FieldXY describes a two dimensional vector field where the
+// X and Y coordinates are arranged on a rectangular grid.
+type FieldXY interface {
+	// Dims returns the dimensions of the grid.
+	Dims() (c, r int)
+
+	// Vector returns the value of a grid value at (c, r).
+	// It will panic if c or r are out of bounds for the grid.
+	Vector(c, r int) XY
+
+	// X returns the coordinate for the column at the index x.
+	// It will panic if c is out of bounds for the grid.
+	X(c int) float64
+
+	// Y returns the coordinate for the row at the index r.
+	// It will panic if r is out of bounds for the grid.
+	Y(r int) float64
+}
+
+// Field implements the Plotter interface, drawing
+// a vector field of the values in the FieldXY field.
+type Field struct {
+	FieldXY FieldXY
+
+	// DrawGlyph is the user hook to draw a field
+	// vector glyph. The function should draw a unit
+	// vector to (1, 0) on the vg.Canvas, c.
+	// The direction and magnitude of v can be used
+	// to determine properties of the glyph drawing
+	// but should not be used to determine size or
+	// directions of the glyph.
+	//
+	// If DrawGlyph is nil, a simple black arrow will
+	// be drawn.
+	DrawGlyph func(c vg.Canvas, v XY)
```

Comment:
or maybe the normalized magnitude.

Reply:
or, actually, providing the row and column indexes as arguments would probably be better, because then the user can pull in whatever external information they need.

Reply:
If the user wants to change the colour based on the magnitude, they can calculate the magnitude from `v`. This was the major reason v was included (which is the normalised vector).

I don't like the idea of a completely free approach to transforms. It opens up whole new world of issues to have to answer. This is simple and works for the vast majority of cases with close to zero work for the user.

I can see some merits in providing the row/column, but I need to think about it some more.

Reply:
I somehow missed the fact that `v` was already included as an argument.